spot_wrapper/wrapper.py

```diff
@@ -212,42 +313,124 @@ def _start_query(self):
-                         not self._spot_wrapper._last_trajectory_command_precise):
+                if (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_AT_GOAL
+                    or (
+                        status
+                        == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_NEAR_GOAL
+                        and not self._spot_wrapper._last_trajectory_command_precise
+                    )
+                ):
                     self._spot_wrapper._at_goal = True
                     # Clear the command once at the goal
                     self._spot_wrapper._last_trajectory_command = None
-                elif status == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_GOING_TO_GOAL:
+                elif (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_GOING_TO_GOAL
+                ):
                     is_moving = True
-                elif status == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_NEAR_GOAL:
+                elif (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_NEAR_GOAL
+                ):
                     is_moving = True
                     self._spot_wrapper._near_goal = True
+                elif (
+                    status
+                    == basic_command_pb2.SE2TrajectoryCommand.Feedback.STATUS_UNKNOWN
+                ):
+                    self._spot_wrapper._trajectory_status_unknown = True
+                    self._spot_wrapper._last_trajectory_command = None
                 else:
+                    self._logger.error(
+                        "Received trajectory command status outside of expected range, value is {}".format(
+                            status
+                        )
+                    )
                     self._spot_wrapper._last_trajectory_command = None
             except (ResponseError, RpcError) as e:
                 self._logger.error("Error when getting robot command feedback: %s", e)
                 self._spot_wrapper._last_trajectory_command = None
 
         self._spot_wrapper._is_moving = is_moving
 
+        # We must check if any command currently has a non-None value for its id. If we don't do this, this stand
+        # command can cause other commands to be interrupted before they get to start
+        if (
+            self._spot_wrapper.is_standing
+            and not self._spot_wrapper.is_moving
+            and self._spot_wrapper._last_trajectory_command is not None
+            and self._spot_wrapper._last_stand_command is not None
+            and self._spot_wrapper._last_velocity_command_time is not None
+            and self._spot_wrapper._last_docking_command is not None
+        ):
+            self._spot_wrapper.stand(False)
+
+
+class AsyncEStopMonitor(AsyncPeriodicQuery):
```

Comment:
My original PR which added this is at https://github.com/heuristicus/spot_ros/pull/72. Part of the reason I added it was to prevent an error which often came up when killing the driver because of an uninitialised variable, but also there was no monitoring of the estop client and I thought it would be wise to include something which checked its status.